unlox_driver:
  ros__parameters:
    online: true           # input from serial port(true) or file(false)
    ## if `online == 1` then following options may be set
    input_serial_port: "/dev/ttyACM0"
    serial_baud_rate: 38400
    input_rtcm: false       # whether input RTCM data to receiver for RTK solution
    rtcm_tcp_port: 3503
    config_receiver_at_start: false     # whether to config receiver during booting, ** NOT SUPPORT YET **
    receiver_config_filepath: "~/ros2_ws/src/ublox_driver/config/ublox_rcv_config.yaml"
    ## if `online == 0` then following options may be set, `serial_baud_rate` is used for speed control
    ubx_filepath: "~/tmp/ublox_driver_test/2020_11_30_7_14_33.ubx"
    rtk_correction_ecef: [ 0.0, 0.0, 0.0 ]   # apply RTK correction in case of the position of base station reported in RTCM is biased

    # output options
    to_ros: true           # publish as ros topic
    to_file: false          # dump data to file
    dump_dir: "~/tmp/ublox_driver_test/"      # dump data to this directory, if applicable
    to_serial: false        # output data to serial port, for debugging purpose
    output_serial_port: "/dev/ttyACM0"                  # dump data to this serial port, if applicable
